src/courses/tests/test_views.py

```diff
@@ -1050,6 +1047,164 @@ def test_no_test_users_in_histogram_values(self):
+        self.course = baker.make(Course)
+
+        # freeze_time doesnt affect "SiteConfig.get().active_semester"
+        # we have to manually set the dates
+        self.semester = SiteConfig.get().active_semester
+        self.semester.first_day = datetime.datetime(2024, 1, 1, tzinfo=datetime.timezone.utc)
+        self.semester.last_day = self.semester.first_day + datetime.timedelta(days=135)
+        self.semester.save()
+
+        self.student_course = baker.make(
+            CourseStudent,
+            user=self.student,
+            semester=self.semester,
+            course=self.course,
+            block=self.block
+        )
+
+    def create_quest_and_submissions(self, xp, quest_submission_date, quest_submission_quantity=1):
+        """
+        Creates and returns quest with linked quest submission objects for self.user
+
+        Args:
+            xp (int): amount of xp quest object will have,
+            quest_submission_date: when the submission is approved
+            quest_submission_quantity (int, optional): how many submissions are created. Defaults to 1.
+
+        Returns:
+            tuple[Quest, list[QuestSubmission]]: tuple of Quest object + list of QuestSubmission objects
+        """
+
+        quest = baker.make('quest_manager.Quest', xp=xp)
+        quest_submissions = baker.make(
+            'quest_manager.QuestSubmission',
+            quest=quest,
+            user=self.student,
+            is_completed=True,
+            is_approved=True,
+            semester=self.semester,
+            time_approved=quest_submission_date,
+            _quantity=quest_submission_quantity,
+        )
+
+        return quest, quest_submissions
+
+    def test_non_ajax_status_code(self):
+        # 302 unless verified ajax POST request
+        self.assertRedirectsLogin('courses:ajax_progress_chart', args=[self.student.pk])
+
+    def test_ajax_status_code_for_anonymous(self):
+        # checks redirect with ajax style request "HTTP_X_REQUESTED_WITH='XMLHttpRequest'"
+        response = self.client.post(reverse('courses:ajax_progress_chart', args=[self.student.pk]), HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEqual(response.status_code, 302)
+
+    def test_ajax_status_code_for_student(self):
+        self.client.force_login(self.student)
+
+        response = self.client.post(reverse('courses:ajax_progress_chart', args=[self.student.pk]), HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEqual(response.status_code, 200)
+
+    def test_ajax_xp_data__correct_xp_current_day(self):
+        """ tests if xp_data from ajax request holds the correct xp on different days of the week.
+        uses freeze_time to test the current day as Fri, Sat, Sun, and Mon
+        """
+        self.client.force_login(self.student)
+
+        # start adding submissions 1 week after
+        # create submissions from mon to fri
+        starting_date = datetime.datetime(2024, 1, 8, tzinfo=datetime.timezone.utc)
```

Comment:
not a sunday its monday
(y/m/d) -> 2024/1/8 -> 1 Jan 2024
![image](https://github.com/bytedeck/bytedeck/assets/39788517/bfff2654-b7e5-485f-bda7-a35f284992c5)

Reply:
added this comment for clarity
```
# start adding submissions 1 week after the first_day in order to prevent any
        # intersection with this bug: https://github.com/bytedeck/bytedeck/issues/1607
```

Reply:
i think i removed it already